From e9c36295b8d7ef35d61bb102a0439036794fc93f Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <icenowy@aosc.io>
Date: Tue, 8 Sep 2026 12:01:16 +0800
Subject: [PATCH 1/2] AOSCOS: fix(term.c): disable cursor style/blink under
 KMSCON

KMSCON says it it's xterm-256color, but it's not.

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 src/term.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/term.c b/src/term.c
index b789f3924..214130bac 100644
--- a/src/term.c
+++ b/src/term.c
@@ -5294,6 +5294,13 @@ handle_version_response(int first, int *arg, int argc, char_u *tp)
 							// sequences
 	}
 
+	// KMSCON sends 1;1;0 and it cannot handle cursor style and blink
+	if (arg[0] == 1 && arg[1] == 1 && arg[2] == 0)
+	{
+	    term_props[TPR_CURSOR_STYLE].tpr_status = TPR_NO;
+	    term_props[TPR_CURSOR_BLINK].tpr_status = TPR_NO;
+	}
+
 	// Xterm first responded to this request at patch level
 	// 95, so assume anything below 95 is not xterm and hopefully supports
 	// the underline RGB color sequence.
-- 
2.55.0

